SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -3526,8 +3526,11 @@ void CSolver::LoadInletProfile(CGeometry **geometry,
 
   auto profile_filename = config->GetInlet_FileName();
 
-  unsigned short nVar_Turb = 0;
-  if (config->GetKind_Turb_Model() != TURB_MODEL::NONE) nVar_Turb = solver[MESH_0][TURB_SOL]->GetnVar();
+  const auto turbulence = config->GetKind_Turb_Model() != TURB_MODEL::NONE;
+  const unsigned short nVar_Turb = turbulence ? solver[MESH_0][TURB_SOL]->GetnVar() : 0;
+
+  const auto species = config->GetKind_Species_Model() != SPECIES_MODEL::NONE;
+  const unsigned short nVar_Species = species ? solver[MESH_0][SPECIES_SOL]->GetnVar() : 0;
```

Comment:
segfault if those solvers do not exist

Reply:
It does not not segfault though if the solvers do not exist ... I would have noticed that (and the backward facing step tutorial as well. I can make it a real if-else block though in case that is safer

Reply:
:+1: you are right, the ternary operator "short circuits"